src/app/components/dashbord-user/announcement-user/announcement-user.component.html

```diff
@@ -39,13 +48,13 @@
             </td>
             <td>
               <button
-                (click)="onViewAnnouncement(announcement.announcementId)"
+                (click)="onViewAnnouncement(announcement.announcementId)" [disabled]="isTerminated(announcement.announcementStatus)"
```

Comment:
should be isFinished et pas isTerminated ou bien tu peux faire tout simplement:
[disabled]="announcement.announcementStatus==='FINISHED'"